pkg/runtimeproxy/server/cri/criserver.go

```diff
@@ -94,45 +95,47 @@ func (c *RuntimeManagerCriServer) getRuntimeHookInfo(serviceType RuntimeServiceT
 
 func (c *RuntimeManagerCriServer) interceptRuntimeRequest(serviceType RuntimeServiceType,
 	ctx context.Context, request interface{}, handler grpc.UnaryHandler) (interface{}, error) {
-
 	runtimeHookPath, runtimeResourceType := c.getRuntimeHookInfo(serviceType)
 	resourceExecutor := resource_executor.NewRuntimeResourceExecutor(runtimeResourceType)
 
-	if err := resourceExecutor.ParseRequest(request); err != nil {
+	callHookOperation, err := resourceExecutor.ParseRequest(request)
+	if err != nil {
 		klog.Errorf("fail to parse request %v %v", request, err)
 	}
 	defer resourceExecutor.DeleteCheckpointIfNeed(request)
 
-	// pre call hook server
-	// TODO deal with the Dispatch response
-	response, err := c.hookDispatcher.Dispatch(ctx, runtimeHookPath, config.PreHook, resourceExecutor.GenerateHookRequest())
-	if err != nil {
-		klog.Errorf("fail to call hook server %v", err)
-	} else if response == nil {
-		// when hook is not registered, the response will become nil
-		klog.Warningf("runtime hook path %s does not register any PreHooks", string(runtimeHookPath))
-	} else {
-		if err = resourceExecutor.UpdateRequest(response, request); err != nil {
-			klog.Errorf("failed to update cri request %v", err)
+	switch callHookOperation {
+	case utils.ShouldCallHookPlugin:
+		// TODO deal with the Dispatch response
+		response, err := c.hookDispatcher.Dispatch(ctx, runtimeHookPath, config.PreHook, resourceExecutor.GenerateHookRequest())
+		if err != nil {
+			klog.Errorf("fail to call hook server %v", err)
+		} else if response == nil {
+			// when hook is not registered, the response will become nil
+			klog.Warningf("runtime hook path %s does not register any PreHooks", string(runtimeHookPath))
```

Comment:
maybe level klog.V(4).Info is enough for no hook registered

Reply:
> maybe level klog.V(4).Info is enough for no hook registered

this log would be removed in next patch